packages/atomic/.storybook/preview-head.html

```diff
@@ -1,3 +1,13 @@
 <script type="module" src="atomic/atomic.esm.js"></script>
 <script nomodule src="atomic/atomic.js"></script>
+<script>
+  window.addEventListener('load', () => {
+    setTimeout(() => {
+      let message = {
+        height: document.body.scrollHeight,
+      };
+      window.top.postMessage(message, '*');
+    }, 1500);
```

Comment:
without a timeout, the `document.body.scrollHeight` value is 32 px, but after the timeout that value becomes relevant.

@kaitlynkatz  and I tried to catch the exact event after which `document.body.scrollHeight` becomes relevant, but eventually we didn't make a lot of success. We observed that the closest prior event that the iframe sends to the parent window is `a11y/extension/search_executed`, but still, the height value WOULD NOT become relevant immediately after the parent window received that event.

Reply:
Yeah, Atomic does not expose any lifecycle hook/event after it has rendered.

It's something we will need to add, eventually, I think. 

It would be helpful for implementers as a generic tool, to get callbacked when components are loaded, rendered, unmounted, etc. 

@ThibodeauJF Do you know if such events are already emitted Stencil, or if we need to roll our own ?

Anyway, for the timeout, I think we could instead use this: https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver

Whenever `document.body` changes, you could emit the new scroll height value. 

That way, if an end user interaction causes a component to grow in size (for example, let's say the end user uses load more feature of a facet), the size would adapt without the need for a scrollbar.

Reply:
@olamothe alright, I'll try to play with MutationObserver a bit, thx! If I don't handle it relatively quickly, can I ask for your help then?

Reply:
Sure !

Reply:
@olamothe 

I've updated the code in both pages.

Works like a charm! 😍

Iframe code:

```js
  window.addEventListener('load', () => {
    const container = document.body;

    const observer = new ResizeObserver(function () {
      let message = {
        height: document.body.scrollHeight,
      };
      window.top.postMessage(message, '*');
    });

    for (let i = 0; i < container.children.length; i++) {
      observer.observe(container.children[i]);
    }
  });
```

Parent window code:

```html
<iframe frameborder="0" width="100%" id="componentIframe" src="http://localhost:6006/iframe.html?id=atomic-categoryfacet--default-category-facet&args=&viewMode=story"></iframe>

<script>

  let iframe = document.querySelector("#componentIframe");

  let previousHeight = 0;

  window.addEventListener('message', (e) => {
   	
    let message = e.data;

    if (message.height !== previousHeight) {
      iframe.style.height = message.height + 10 + 'px';
      previousHeight = message.height;
    }

  } , false);

</script>
```

Reply:
:gg: !